### ISSUE TITLE
View Site: Private sites with custom domain redirect to device browser with error

### ISSUE THREAD

**Issue by Robertght** [wordpress-mobile/WordPress-Android#10480]
### Expected behavior
Viewing a private site with a custom domain should show the user a logged-in view in the in-app browser.

### Actual behavior
The in-app browser redirects to the device browser with an error:

![image](https://user-images.githubusercontent.com/6737345/64231817-3ef60580-cef9-11e9-8ab3-0e88cee43613.png)

### Steps to reproduce the behavior
1. Make sure you have a WordPress.com site set to private with a custom domain.
2. Open it in a browser where you are logged in - it should show up the private site page.
3. You'll want to set another browser as the main one where you are not logged in. Tap on View site - the first time it will show up the private page. Go back to the WordPress app and click on "View site" again, this time you'll see the Expected behavior screen. 
4. Doing the same previous step after seeing the error, will show up the Private page.

##### Tested on Huawei and Samsung Android 7 and 9.0.1, WPAndroid Alpha188

Also confirmed by a user, using Nokia 7 with Android 9.0 and Chrome 76 (asking the user for the WordPress app version)

**Comment by blazevivan** [comment]
From the user: application version is 13.1

**Comment by rachelmcr** [comment]
Thanks for the report! I tested and updated the issue title and expected/actual behavior based on what I found in testing. I'm seeing different steps to reproduce than in the original report, so I'll share my steps here:

1. Make sure you have a WordPress.com site set to private with a custom domain.
2. Open the site in the device browser where you are logged in; the site loads.
3. Log out of WordPress.com and open the site in the device browser; you see a login page.
4. In the app, select "View Site" for the same site.

Result: The app starts to open the site in the in-app browser but then redirects to the device browser with the error shown in the screenshot above.
